### PR TITLE
Km sprint 2 fetch

### PR DESCRIPTION
**Pull request by Kristiannmiller**
## Description
This PR:

- Adds a promise.all function to our fetch request functions in order to assure that all promises are met
- Adds a handler function for updating the data model before invoking dom functions
- Changes some global variable names to be more descriptive and empathetic
- Organizes some functions within scripts.js for readability

## Motivation and Context
A promise.all function was needed to ensure that all promises are met before loading other functions. 
The handler function was needed for single responsibility of functions gathering data
The global variables were named in a confusing manner, which made understanding the original codebase difficult

## How Has This Been Tested?
This has been tested using the console in the browser and the local host site

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
My change requires a change to the documentation.
I have updated the documentation accordingly.
I have added tests to cover my changes.
All new and existing tests passed.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -1,45 +1,44 @@
...
 let savedRecipesBtn = document.querySelector(".saved-recipes-btn");
 let searchBtn = document.querySelector(".search-btn");
 let searchForm = document.querySelector("#search");
 let searchInput = document.querySelector("#search-input");
 let showPantryRecipes = document.querySelector(".show-pantry-recipes-btn");
 let tagList = document.querySelector(".tag-list");
-let user;
-let users;
+
+// ************ GLOBAL VARIABLES ***************
```
Comment (ericberglund117) [inline src/scripts.js:26]: Reorganizing the data like this in scripts is much easier to read now. Thank you!

File: src/scripts.js [hunk 2]
```diff
@@ -50,38 +49,53 @@ showPantryRecipes.addEventListener("click", findCheckedPantryBoxes);
...
 
+function loadPageInfo(allData) {
+  allUsersData = allData[0]
+  ingredientsData = allData[1]
+  recipeData = allData[2]
+  generateUser(allUsersData)
+  findPantryInfo(ingredientsData)
+  findTags(recipeData)
+  createCards(recipeData)
+}
```
Comment (ericberglund117) [inline src/scripts.js:87]: Super pumped about the success of Promise. all and using this function to load the data. Way to go!

**Review by ericberglund117 — APPROVED** [review]

File: src/scripts.js (hunk 1 above)
Comment (chadrick-d-dev) [reply src/scripts.js:26]: Yeah I like this new section, great addition for organization.

File: src/scripts.js [hunk 3]
```diff
@@ -50,38 +49,53 @@ showPantryRecipes.addEventListener("click", findCheckedPantryBoxes);
...
 function getRecipes() {
-  fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/recipes/recipeData')
+  return fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/recipes/recipeData')
     .then(response => response.json())
-    .then(data => {recipeData = data.recipeData, findTags(recipeData), createCards(recipeData)})
+    .then(data => data.recipeData)
     .catch(error => console.log(error))
 }
 
+function loadPageInfo(allData) {
```
Comment (chadrick-d-dev) [inline src/scripts.js:79]: I like this function a lot. Great way to get all the information plugged in.

**Review by chadrick-d-dev — COMMENTED** [review]
Check out my comments!

**Review by chadrick-d-dev — APPROVED** [review]

**Pull request merged**